lib/internal/http2/util.js

```diff
@@ -349,10 +354,19 @@ function updateSettingsBuffer(settings) {
     settingsBuffer[IDX_SETTINGS_MAX_FRAME_SIZE] =
       settings.maxFrameSize;
   }
-  if (typeof settings.maxHeaderListSize === 'number') {
+  if (typeof settings.maxHeaderListSize === 'number' ||
+      typeof settings.maxHeaderSize === 'number') {
     flags |= (1 << IDX_SETTINGS_MAX_HEADER_LIST_SIZE);
+    if (settings.maxHeaderSize && settings.maxHeaderListSize) {
+      if (settings.maxHeaderSize !== settings.maxHeaderListSize) {
+        throw new ERR_HTTP2_INVALID_SETTING_VALUE.RangeError(
+          'maxHeaderListSize', settings.maxHeaderListSize
```

Comment:
@Himself65 should we be more explicit about this? I think this somehow fails to convey the exact message of why this was thrown in the first place to the user (`maxHeaderSize !== maxHeaderListSize`). Maybe we should add a new error code here?

Reply:
I think `maxHeaderSize` is an alias of `maxHeaderListSize`, so throw an error if they are not the same isn't a good idea.

I personally would like to `emitWarning("'maxHeaderListSize' overwrite the 'maxHeaderSize'")`(for backport, so `ListSize`  have the higher priority than `HeaderSize`)

Reply:
What about other Collaborators have any better options on this?